src/optable.ts

```diff
@@ -909,16 +923,22 @@ optable[opcodes.STORE_MAP] = function(f: Py_FrameObject) {
     d.set(key, val);
 }
 
-function setup_block(f: Py_FrameObject, op: number) {
+function setup_block(f: Py_FrameObject, t: Thread, op: number) {
     var delta = f.readArg();
     // push a block to the block stack
     var stackSize = f.stack.length;
     var loopPos = f.lastInst;
     f.blockStack.push([stackSize, loopPos, loopPos+delta, op]);
+
+    // For handling StopIteration exceptions inside of for loops, generated by exhausted generators
+    if (op === opcodes.SETUP_LOOP) {
+        t.loop_exc_block.push([stackSize, loopPos, loopPos + delta, opcodes.SETUP_EXCEPT]);
```

Comment:
OK, I see, it is during a return value. Looks like these are transferred to the block stack. How is this different from the block stack?